src/HexagonOptimize.cpp

```diff
@@ -260,6 +263,27 @@ bool process_match_flags(vector<Expr> &matches, int flags) {
         internal_assert(matches.size() >= 3);
         std::swap(matches[1], matches[2]);
     }
+    if (flags & Pattern::SafeReinterpretOp0) {
+        // Use bounds inference to check if the first operand can
+        // be safely reinterpreted.
+        const Expr &expr = matches[0];
+        const Type &t = expr.type();
+        if (t.is_int()) {
+            // A signed integer can be reinterpreted as unsigned if strictly positive.
+            // TODO: might want to keep track of scope of bounds information.
+            const ConstantInterval ibounds = constant_integer_bounds(expr);
+            const Type reint_type = UInt(t.bits());
+            return ibounds.min_defined && reint_type.can_represent(ibounds.min);
```

Comment:
Same comment here. Use `reint_type.can_represent(ibounds)` (or just `reint_type.can_represent(constant_integer_bounds(expr))`) so the reader doesn't have to think about whether the max needs checking.